presto-main/src/main/java/com/facebook/presto/cost/ComparisonStatsCalculator.java

```diff
@@ -27,53 +28,64 @@
 import static java.lang.Double.POSITIVE_INFINITY;
 import static java.lang.Double.isNaN;
 
-public final class ComparisonStatsCalculator
+public class ComparisonStatsCalculator
 {
-    private ComparisonStatsCalculator() {}
+    private ComparisonStatsCalculator()
+    {}
 
-    public static PlanNodeStatsEstimate comparisonSymbolToLiteralStats(
+    public static PlanNodeStatsEstimate comparisonExpressionToLiteralStats(
             PlanNodeStatsEstimate inputStatistics,
-            Symbol symbol,
+            Optional<Symbol> symbol,
+            SymbolStatsEstimate expressionStats,
```

Comment:
Name is a bit unfortunate as it is symbol stats if symbol is present or expression stats if symbol is absent.

Also I do not like this approach, what would you say for (two suggested approaches):
 -  from methods like `expressionToLiteralEquality` you could return tuple of new row count and new expression symbol stats, then in this method you could use that information to return plan node stats with updated row count and to new symbol stats
 - you could have artificial symbol for the expression stats inserted into plan node stats

Reply:
Symbol is also an expression